internal/connect/command_offset_update.go

```diff
@@ -0,0 +1,142 @@
+	"github.com/confluentinc/cli/v3/pkg/errors"
+	"github.com/confluentinc/cli/v3/pkg/examples"
+	"github.com/confluentinc/cli/v3/pkg/kafka"
+	"github.com/confluentinc/cli/v3/pkg/output"
+	"github.com/confluentinc/cli/v3/pkg/retry"
+)
+
+func (c *offsetCommand) newUpdateCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:               "update <id>",
+		Short:             "Update a connector's offsets.",
+		Args:              cobra.ExactArgs(1),
+		RunE:              c.update,
+		ValidArgsFunction: pcmd.NewValidArgsFunction(c.validArgs),
+		Example: examples.BuildExampleString(
+			examples.Example{
+				Text: "Update offsets for a connector in the current or specified Kafka cluster context.",
+				Code: "confluent connect offset update lcc-123456 --config-file config.json",
+			},
+			examples.Example{
+				Code: "confluent connect offset update lcc-123456 --config-file config.json --cluster lkc-123456",
+			},
+		),
+	}
+
+	cmd.Flags().String("config-file", "", "JSON file containing connector offsets to set to.")
+	pcmd.AddClusterFlag(cmd, c.AuthenticatedCLICommand)
+	pcmd.AddContextFlag(cmd, c.CLICommand)
+	pcmd.AddEnvironmentFlag(cmd, c.AuthenticatedCLICommand)
+	pcmd.AddOutputFlag(cmd)
+
+	cobra.CheckErr(cmd.MarkFlagFilename("config-file", "json"))
+	cobra.CheckErr(cmd.MarkFlagRequired("config-file"))
+
+	return cmd
+}
+
+func (c *offsetCommand) update(cmd *cobra.Command, args []string) error {
+	kafkaCluster, err := kafka.GetClusterForCommand(c.V2Client, c.Context)
+	if err != nil {
+		return err
+	}
+
+	environmentId, err := c.Context.EnvironmentId()
+	if err != nil {
+		return err
+	}
+
+	connector, err := c.V2Client.GetConnectorExpansionById(args[0], environmentId, kafkaCluster.ID)
+	if err != nil {
+		return err
+	}
+
+	connectorName := connector.Info.GetName()
+	configFile, err := cmd.Flags().GetString("config-file")
+	if err != nil {
+		return err
+	}
+
+	request, err := c.getAlterOffsetRequestBody(configFile)
+	if err != nil {
+		return err
+	}
+
+	_, err = c.V2Client.AlterConnectorOffsets(connectorName, environmentId, kafkaCluster.ID, *request)
+	if err != nil {
+		return err
+	}
+
+	var offsetStatus connectv1.ConnectV1AlterOffsetStatus
+
+	err = retry.Retry(time.Second, 30*time.Second, func() error {
+		offsetStatus, err = c.V2Client.AlterConnectorOffsetsRequestStatus(connectorName, environmentId, kafkaCluster.ID)
+		if err != nil {
+			return err
+		}
+
+		if offsetStatus.GetStatus().Phase != "PENDING" {
+			return nil
+		}
+		return fmt.Errorf("update offset request still pending, checking status again")
+	})
+
+	var appliedAt string
+	if offsetStatus.AppliedAt.IsSet() {
+		appliedAt = offsetStatus.AppliedAt.Get().String()
+	}
+
+	var phase string
+	var message string
+	_, isStatusSet := offsetStatus.GetStatusOk()
+	if isStatusSet {
+		phase = offsetStatus.Status.Phase
+		if len(phase) == 0 {
+			phase = "PENDING"
+		}
+		if messagePtr := offsetStatus.Status.Message; messagePtr != nil {
+			message = *messagePtr
+		}
+	}
```

Comment:
I don't think we need to do all this, if the offset response is still pending after 30second the response should have all the required fields populated i.e. Phase should be PENDING and so on. Let's not add processing logic in the CLI, it should just show the API response as close as possible.

Reply:
Similar for the processing done in status code for CLI.

Reply:
This was done if the offset status command doesnt return within 30 sec.

Reply:
Removed